purchase_location_by_line/views/purchase_view.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="/form/sheet/notebook/page/field/tree/field[@name='date_planned']" position="after">
+                <field name="location_dest_id"/>
```

Comment:
@mreficent There is an inconsistency here but maybe it's wanted.
On your purchase order, you have the picking type (Deliver To) field which belongs to a warehouse.
You have to define a domain here to restrict location selection according to that 'Deliver To' field.

Reply:
It's wanted, and you should consider that a domain is already defined as <code>[('usage', 'in', ['internal', 'transit'])]</code>.

Reply:
@mreficent I saw that one.

What I wanted to say is there can be an inconsistency between the location**s** defined on each purchase order line and the 'Deliver To' field on purchase order.

It can be weird for user to have 'Deliver To' : 'Chicago/Stock' and one line with 'Chicago/Stock' and one line with 'WH/Stock' for example (I reproduced it on runbot) as both locations are in different warehouses...

Reply:
Yes, but that's on purpose. The usecase would be for instance a PO with lines that must be sent to 'Location/receptions' and 'Different_parent_locations/office'. It can be confusing, that's right, but it's the better approach we found last time.

Reply:
Ok, I understand. This should be clearer in documentation (as I understood first it was to dispatch products on several different children locations - like putaway strategies).

Reply:
@mreficent please add some lines to describe this in the documentation as suggested, I agree that might be a little confusing.

Reply:
Done